python/geoipsets/maxmind.py

```diff
@@ -160,7 +168,9 @@ def build_sets(self, country_code_map: dict, zip_ref: ZipFile, dir_prefix: str,
                         ipset_file = ipset_dir / set_name
                         if not ipset_file.is_file():
                             with open(ipset_file, 'a') as f:
-                                f.write("create " + set_name + " hash:net " + inet_family + " maxelem 131072 comment\n")
+                                # round up to the next next power of 2 for a load ratio < 0.5
```

Comment:
I'm not sure _maxelem_ has any bearing on the actual size of the underlying hash data structure used by _ipset_ -- and therefore any effect on the load ratio of the hash table. It is simply a cap on the number of elements in a set.

There is a separate _hashsize_ setting that controls the initial size of the hash table, but the actual size is managed internally by _ipset_ based on the number of set elements at any given time. (According to the [manpage](https://ipset.netfilter.org/ipset.man.html) the size is doubled if existing size is exhausted but there is probably more to it than that.)

Given this, I think setting _maxelem_ to the next power of 2 (instead of the next next power of 2) is sufficient. If you agree, please push that change to the PR. If I've missed something, though, please do let me know.

Reply:
You are absolutely right. Now begging the question: Do we set the exact `maxelem` since it doesn't need to be a power of 2 
 according to the man page?

Reply:
Let's stick with a power of 2. The default (65536) and example (2048) for _maxelem_ in the manpage are both a power of 2 and doing otherwise just feels wrong.